src/functions-reference/hidden_markov_models.Rmd

```diff
@@ -0,0 +1,69 @@
+# Hidden Markov Models (with discrete hidden variables)
+
+An elementary first-order Hidden Markov model 
+is a probabilistic model over
+$N$ observations, $y_n$, and $N$ hidden states, $x_n$,
+which can be fully defined by the conditional distributions
+$p(y_n \mid x_n)$ and $p(x_n \mid x_{n - 1})$.
+When $x$ is continuous, the user can explicitly encode these distributions
+in Stan and use Markov chain Monte Carlo to integrate $x$ out.
+
+When each state $x$ takes a value over a discrete and finite set, say $\{1, 2, ..., K\}$,
+we can take advantage of the dependency structure
+to marginalize $x$ and compute $p(y)$.
```

Comment:
[optional] One thing that might be confusing here is that when you say $p(y)$ you really mean $p(y | \theta)$ where $\theta$ is anything else that's used to compute any of the input quantities. @wds15 suggested for the ODE stuff we be explicit about the $\theta$ dependence in everything. You can see what we did here: https://mc-stan.org/docs/2_24/functions-reference/functions-ode-solver.html . You can decide whether or not this would be useful here.

Reply:
Excellent idea.